fem/tmop.hpp

```diff
@@ -69,6 +73,20 @@ class TMOP_QualityMetric : public HyperelasticModel
    virtual void EvalPW(const DenseMatrix &Jpt, DenseMatrix &PW) const
    { PW = 0.0;}
 
+   // First-derivative hook for AD-based computations:
```

Comment:
We should stress in the comments that this function is internal, for AD connection purposes. I don't like it, but also couldn't figure a way to remove it from the `public`. Same for `EvalW_AD2`.